pageserver/src/tenant/size.rs

```diff
@@ -44,6 +44,129 @@ struct TimelineInputs {
     next_gc_cutoff: Lsn,
 }
 
+// Adjust BracnchFrom sorting so that we always process ancestor
+// before descendants.
+//
+// i.e. if we have following order
+// Update { lsn: 0/0, command: BranchFrom(None), timeline_id: 1 },
+// Update { lsn: 0/169AD58, command: Update(25387008), timeline_id: 1 },
+// Update { lsn: 0/169AD58, command: BranchFrom(Some(2)), timeline_id: 3 },
+// Update { lsn: 0/169AD58, command: BranchFrom(Some(1)), timeline_id: 2 },
+//
+// last two lines must be reordered to
+// Update { lsn: 0/169AD58, command: BranchFrom(Some(1)), timeline_id: 2 },
+// Update { lsn: 0/169AD58, command: BranchFrom(Some(2)), timeline_id: 3 },
+//
+fn sort_updates_in_tree_order(updates: Vec<Update>) -> anyhow::Result<Vec<Update>> {
```

Comment:
It is not explained why do we need to sort updates in tree order...

Reply:
I think the new fn comment now explains, and the test case has been added, so resolving this.